hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientMembershipListener.java

```diff
@@ -112,13 +113,14 @@ public void handle(Collection<Member> initialMembers) {
 
     @Override
     public void handle(String uuid, String key, int opType, String value) {
-        Collection<Member> members = clusterService.getMemberList();
-        for (Member target : members) {
+        Collection<Member> memberList = clusterService.getMemberList();
+        for (Member target : memberList) {
             if (target.getUuid().equals(uuid)) {
                 final MemberAttributeOperationType operationType = MemberAttributeOperationType.getValue(opType);
                 ((AbstractMember) target).updateAttribute(operationType, key, value);
                 MemberAttributeEvent memberAttributeEvent =
-                        new MemberAttributeEvent(client.getCluster(), target, operationType, key, value);
+                        new MemberAttributeEvent(client.getCluster(), target, operationType, key, value,
+                                unmodifiableSet(members));
```

Comment:
I'm not too sure about a threading model here. However I noticed other events are referring to `members` without any synchronization -> I assume it's ok here too.